docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,215 @@
+
+### Manipulating the time of the network
+
+We are deploying our `Lock` contract with an unlock time of one year. If we want to write a test that checks what happens after the unlock time has passed, we can’t wait that amount of time. We could use a shorter amount, like 5 seconds, but that’s a less realistic value and it's still a long time to wait in a test.
+
+The solution is to simulate the passage of time. This can be done with the [`time.increaseTo`](</hardhat-network-helpers/docs/reference#increaseto(timestamp)>) network helper, which mines a new block with the given timestamp:
+
+```tsx
+it("Should transfer the funds to the owner", async function () {
+  // ...deploy the contract...
+
+  await time.increaseTo(unlockTime);
+
+  // this will throw if the transaction reverts
+  await lock.withdraw();
+});
+```
+
+As we mentioned, calling `lock.widthdraw()` returns a Promise. If the transaction fails, the promise will be rejected. Using `await` will throw in that case, so the test will fail if the transaction reverts.
+
+### Using a different address
+
+The second check done by the `withdraw` function is that the function was called by the owner of the contract. In these tests, all the deployments and function calls are done by default with the first [configured account](/hardhat-network/docs/reference.md#accounts). If we want to check that only the owner can call some function, we need to use a different account and verify that it fails.
+
+The `ethers.getSigners()` returns an array with all the configured accounts. We can then use the `.connect` method of the contract to call the function with a different account and check that the transaction reverts:
+
+```tsx
+it("Should revert with the right error if called from another account", async function () {
+  // ...deploy the contract...
+
+  const [owner, otherAccount] = await ethers.getSigners();
+
+  // we increase the time of the chain to pass the first check
+  await time.increaseTo(unlockTime);
+
+  // We use lock.connect() to send a transaction from another account
+  await expect(lock.connect(otherAccount).withdraw()).to.be.revertedWith(
+    "You aren't the owner"
+  );
+});
+```
+
+Here again we are calling a function and asserting that it reverts with the correct reason string. The difference is that we are using `.connect(anotherAccount)` to call the method from a different address.
+
+### Using fixtures
+
+So far we've deployed the `Lock` contract in each test. This means that at the beginning of each test we have to get the contract factory and then deploy the contract. This might be fine for a single contract but, if you have a more complicated setup, each test will have several lines at the begginning just to set up the desired state, and most of the time these lines will be the same.
+
+In Mocha, this duplication of code is handled with a `beforeEach` hook:
+
+```tsx
+let lock;
+let unlockTime;
+let lockedAmount = 1_000_000_000;
+beforeEach(async function () {
```

Comment:
just for complete accuracy, shouldn't there be a `describe` block here that contains both the `beforeEach` and the `it`?

Reply:
Done in 80c22a60f51e341f8a39613a180ceadad892fe20, thanks